doc/pedantic_build.sh

```diff
@@ -21,25 +21,30 @@
 # be permissive for warnings related to them
 IGNORE_PATTERN="devlib|bart|wa|exekall.engine"
 
-make SPHINXOPTS='-n' html 2>doc_build.log
-all_warns=$(cat doc_build.log)
-warns=$(cat doc_build.log | grep WARNING | grep -E -v "$IGNORE_PATTERN")
+make SPHINXOPTS='-n --no-color' html 2>doc_build.log
```

Comment:
Why use `--no-color` and then `grep --color=always` ?

Reply:
When letting sphinx-build use colors and then cutting some lines, we end up with the escape sequence to enable red, but we cut out the sequence to go back to normal color. Also, if sphinx-build gives a warning line that is all red, we cannot see easily what grep has selected. By doing that, we can see why a warning is selected or not, so it's easy to spot wrong matches like "wa" unexpectedly matching "was".